ncrypt/crypt.c

```diff
@@ -266,31 +266,29 @@ int mutt_protect(struct Email *e, char *keylist, bool postpone)
       mutt_addr_free(&from);
   }
 
-  const bool c_crypt_protected_headers_write = cs_subset_bool(NeoMutt->sub, "crypt_protected_headers_write");
-  if (c_crypt_protected_headers_write)
-  {
-    struct Envelope *protected_headers = mutt_env_new();
-    mutt_env_set_subject(protected_headers, e->env->subject);
-    /* Note: if other headers get added, such as to, cc, then a call to
-     * mutt_env_to_intl() will need to be added here too. */
-    mutt_prepare_envelope(protected_headers, 0, NeoMutt->sub);
-
-    mutt_env_free(&e->body->mime_headers);
-    e->body->mime_headers = protected_headers;
-    mutt_param_set(&e->body->parameter, "protected-headers", "v1");
-  }
+  struct Envelope *protected_headers = mutt_env_new();
+  mutt_env_set_subject(protected_headers, e->env->subject);
+  mutt_list_copy_tail(&protected_headers->in_reply_to, &e->env->in_reply_to);
+  mutt_addrlist_copy(&protected_headers->from, &e->env->from, false);
+  mutt_addrlist_copy(&protected_headers->to, &e->env->to, false);
+  mutt_addrlist_copy(&protected_headers->cc, &e->env->cc, false);
+  mutt_addrlist_copy(&protected_headers->sender, &e->env->sender, false);
+  mutt_addrlist_copy(&protected_headers->reply_to, &e->env->reply_to, false);
+  mutt_addrlist_copy(&protected_headers->mail_followup_to, &e->env->mail_followup_to, false);
+  mutt_addrlist_copy(&protected_headers->x_original_to, &e->env->x_original_to, false);
+  mutt_env_to_intl(protected_headers, NULL, NULL);
+  mutt_prepare_envelope(protected_headers, 0, NeoMutt->sub);
+
+  mutt_env_free(&e->body->mime_headers);
+  e->body->mime_headers = protected_headers;
+  mutt_param_set(&e->body->parameter, "protected-headers", "v1");
 
 #ifdef USE_AUTOCRYPT
   /* A note about e->body->mime_headers.  If postpone or send
    * fails, the mime_headers is cleared out before returning to the
    * compose menu.  So despite the "robustness" code above and in the
    * gen_gossip_list function below, mime_headers will not be set when
    * entering mutt_protect().
-   *
-   * This is important to note because the user could toggle
-   * $crypt_protected_headers_write or $autocrypt off back in the
-   * compose menu.  We don't want mutt_rfc822_write_header() to write
-   * stale data from one option if the other is set.
    */
   const bool c_autocrypt = cs_subset_bool(NeoMutt->sub, "autocrypt");
   if (c_autocrypt && !postpone && (security & SEC_AUTOCRYPT))
```

Comment:
@flatcap I'm not sure if this means we can further simlify now that the variable has been removed.  Could you please have a look at it?

Reply:
Or maybe rephrase the comment instead of removing it.